gnucash/gnome-utils/gnc-tree-model-owner.c

```diff
@@ -1004,6 +1010,12 @@ gnc_tree_model_owner_event_handler (QofInstance *entity,
             DEBUG("can't generate path");
             break;
         }
+
+        g_assert_not_reached ();
+        /* above disables. untested; removing an owner is not allowed
+           in UI */
+        priv->owner_list = g_list_remove (priv->owner_list, &owner);
+        gncOwnerFree (&owner);
```

Comment:
This part is not tested live, hence has `g_assert_not_reached`. Otherwise I think this branch is safe?

Reply:
No, deliberately crashing isn't safe. Besides, `assert`s are usually disabled in release builds so they're only useful as part of comprehensive testing.

Next, it's nonsensical to assert that a `switch case` isn't ever used and also provide implementation code for it.

Is the GtkTreeModel really the owner of the gncOwner? gncOwner is a subclass of QofInstance and most of those are owned by the QofCollections in qofid.c and are supposed to exist for the duration of the session unless the user explicitly deletes them. A QOF_REMOVE event suggests that that has happened, in which case the GtkTreeModel should remove its pointer to the gncOwner from the list as that object will soon be freed but shouldn't free it; that will be taken care of once the event processing is finished.

Reply:
Agree pointless to do `g_list_remove` since the whole `owner_list` will be freed properly when the session is closed.